bchrpc/bchrpc.proto

```diff
@@ -683,7 +803,152 @@ message TransactionFilter {
     // Filter by data elements contained in pubkey scripts.
     repeated bytes data_elements = 3;
 
-    // Subscribe/Unsubscribe to everything. Other filters
-    // will be ignored.
+    // Subscribed/Unsubscribe to everything. Other filters
+    // will be ignored.    
     bool all_transactions = 4;
+
+    // Subscribed/Unsubscribe to everything SLP. Other filters
+    // will be ignored, except this filter will be overriden by all_transactions=true
+    bool all_slp_transactions = 5;
+
+    // only transactions associated with the included tokenIds
+    repeated bytes slp_token_ids = 6;
+}
+
+// SLP token info used in transaction inputs / outputs
+//
+// WARNING: Some languages (e.g., JavaScript) may not properly handle the 'uint64'
+// for large amounts. For this reason, an annotation has been added for JS to
+// return a string for the amount field instead of casting uint64 to the JS 'number'
+// type. Other languages may require similar treatment.
+//
+message SlpToken {
+    bytes token_id = 1;
+    uint64 amount = 2 [jstype = JS_STRING];
+    bool is_mint_baton = 3;
+    string address = 4;
+    uint32 decimals = 5;
+    SlpAction slp_action = 6;
+    uint32 token_type = 7;
+}
+
+message SlpTransactionInfo {
+    SlpAction slp_action = 1;
+    enum ValidityJudgement {
+        UNKNOWN_OR_INVALID = 0;
+        VALID = 1;
+    }
+    ValidityJudgement validity_judgement = 2;
+    string parse_error = 3;
+    bytes token_id = 4;
+    enum BurnFlags {
+        BURNED_INPUTS_OUTPUTS_TOO_HIGH = 0;
+        BURNED_INPUTS_BAD_OPRETURN = 1;
+        BURNED_INPUTS_OTHER_TOKEN = 2;
+        BURNED_OUTPUTS_MISSING_BCH_VOUT = 3;
+        BURNED_INPUTS_GREATER_THAN_OUTPUTS = 4;
+    }
+    repeated BurnFlags burn_flags = 5;
+    oneof tx_metadata {
+        SlpV1GenesisMetadata v1_genesis = 6;    // NFT1 Group also uses this
+        SlpV1MintMetadata v1_mint = 7;          // NFT1 Group also uses this
+        SlpV1SendMetadata v1_send = 8;          // NFT1 Group also uses this
+        SlpNft1ChildGenesisMetadata nft1_child_genesis = 9;
+        SlpNft1ChildSendMetadata nft1_child_send = 10;
+    }
+}
+
+message SlpV1GenesisMetadata {
+    bytes name = 1;
+    bytes ticker = 2;
+    bytes document_url = 3;
+    bytes document_hash = 4;
+    uint32 decimals = 5;
+    uint32 mint_baton_vout = 6;
+    uint64 mint_amount = 7 [jstype = JS_STRING];
+}
+
+message SlpV1MintMetadata {
+    uint32 mint_baton_vout = 1;
+    uint64 mint_amount = 2 [jstype = JS_STRING];
+}
+
+message SlpV1SendMetadata {
+    repeated uint64 amounts = 1 [jstype = JS_STRING];
+}
+
+message SlpNft1ChildGenesisMetadata {
+    bytes name = 1;
+    bytes ticker = 2;
+    bytes document_url = 3;
+    bytes document_hash = 4;
+    uint32 decimals = 5;
+    bytes group_token_id = 6;
+}
+
+message SlpNft1ChildSendMetadata {
+    bytes group_token_id = 1;
+}
+
+enum SlpAction {       // everything except "NON_SLP" could be annotated with burn_flags  
```

Comment:
Would prefer these comments to be before the enum declaration.

Reply:
done